src/node/mod.rs

```diff
@@ -0,0 +1,1241 @@
+
+    // Transition from Bootstrapping to Joining
+    fn join(&mut self, params: JoinParams) {
+        let JoinParams {
+            network_cfg,
+            elders_info,
+            relocate_payload,
+        } = params;
+
+        self.stage = Stage::Joining(Joining::new(
+            &mut self.core,
+            network_cfg,
+            elders_info,
+            relocate_payload,
+        ));
+    }
+
+    // Transition from Joining to Approved
+    fn approve(
+        &mut self,
+        network_cfg: NetworkParams,
+        connect_type: Connected,
+        gen_pfx_info: GenesisPfxInfo,
+    ) {
+        info!(
+            "This node has been approved to join the network at {:?}!",
+            gen_pfx_info.elders_info.prefix(),
+        );
+
+        let stage = Approved::new(&mut self.core, network_cfg, gen_pfx_info, None);
+        self.stage = Stage::Approved(stage);
+        let _ = self.user_event_tx.send(Event::Connected(connect_type));
+    }
+
+    // Transition from Approved to Bootstrapping on relocation
+    fn relocate(&mut self, params: RelocateParams) {
+        let RelocateParams {
+            network_cfg,
+            conn_infos,
+            details,
+        } = params;
+
+        let mut stage = Bootstrapping::new(network_cfg, Some(details));
+
+        for conn_info in conn_infos {
+            stage.send_bootstrap_request(&mut self.core, conn_info)
+        }
+
+        self.stage = Stage::Bootstrapping(stage);
+    }
+
+    // Transition from Joining to Bootstrapping on join failure
+    fn rebootstrap(&mut self, network_cfg: NetworkParams) {
+        // TODO: preserve relocation details
+        self.stage = Stage::Bootstrapping(Bootstrapping::new(network_cfg, None));
+        self.core.full_id = FullId::gen(&mut self.core.rng);
+        self.core.transport.bootstrap();
+    }
+
+    fn set_log_ident(&self) -> log_utils::Guard {
+        use std::fmt::Write;
+        log_utils::set_ident(|buffer| match &self.stage {
+            Stage::Bootstrapping(_) => write!(buffer, "Bootstrapping({}) ", self.name()),
+            Stage::Joining(stage) => write!(
+                buffer,
+                "Joining({}({:b})) ",
+                self.name(),
+                stage.target_section_prefix()
+            ),
+            Stage::Approved(stage) if !stage.chain.is_self_elder() => write!(
+                buffer,
+                "Adult({}({:b})) ",
+                self.core.name(),
+                stage.chain.our_prefix()
+            ),
+            Stage::Approved(stage) => write!(
+                buffer,
+                "Elder({}({:b})) ",
+                self.core.name(),
+                stage.chain.our_prefix()
+            ),
+            Stage::Terminated => write!(buffer, "Terminated"),
+        })
+    }
+}
+
+#[cfg(feature = "mock_base")]
+impl Node {
+    /// Returns whether the node is approved member of a section.
+    pub fn is_approved(&self) -> bool {
```

Comment:
Good points. There are historic reasons for this terminology, because we used to have this thing called resource proof which every node had to go through before joining a section. There is no "Rejected" currently. So it's probably confusing and we should be using just `is_member` or `joined` (also the `Approved` stage should be called `Joined` or something). I would address this in a followup PR.